[Bug][PodGroup-Controller] panic when podGroup.spec.minMember=0 and no selected pod

### Area

- [ ] Scheduler
- [x] Controller
- [ ] Helm Chart
- [ ] Documents

### Other components

_No response_

### What happened?

PodGroupController panic when podGroup.spec.minMember = 0 nad no selected pod.
```
I1205 07:59:50.510842       1 controller.go:115]  "msg"="Observed a panic in reconciler: runtime error: index out of range [0] with length 0" "PodGroup"={"name":"one-pod-not-changed-001","namespace":"default"} "controller"="podgroup" "controllerGroup"="scheduling.x-k8s.io" "controllerKind"="PodGroup" "name"="one-pod-not-changed-001" "namespace"="default" "reconcileID"="dd2aa431-b753-4f6d-9ae9-f3c30d2d01e3"
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 141 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:116 +0x1a4
panic({0x12d6de0, 0x4000148228})
	/usr/local/go/src/runtime/panic.go:884 +0x1f4
sigs.k8s.io/scheduler-plugins/pkg/controllers.(*PodGroupReconciler).Reconcile(0x40003ecb40, {0x1651760, 0x40004d4270}, {{{0x4000047fb0?, 0x30?}, {0x40000b6b28?, 0x400005bd38?}}})
	/go/src/sigs.k8s.io/scheduler-plugins/pkg/controllers/podgroup_controller.go:110 +0x8dc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1654058?, {0x1651760?, 0x40004d4270?}, {{{0x4000047fb0?, 0xb?}, {0x40000b6b28?, 0x0?}}})
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:119 +0x8c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0x40003725a0, {0x16516b8, 0x40003ecaa0}, {0x12583e0?, 0x400007c000?})
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:316 +0x30c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0x40003725a0, {0x16516b8, 0x40003ecaa0})
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266 +0x1a0
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227 +0x74
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/src/sigs.k8s.io/scheduler-plugins/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:223 +0x43c
```

### What did you expect to happen?

PodGroup status phase shoule be `scheduling`

### How can we reproduce it (as minimally and precisely as possible)?

Create a podGroup with minMember=0 and not create any pod.

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
Client Version: v1.30.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.2
```

</details>


### Scheduler Plugins version

<details>
v0.28.9 (latest version also have same code)
</details>

--- a/pkg/controllers/podgroup_controller.go
+++ b/pkg/controllers/podgroup_controller.go
@@ -107,7 +107,9 @@ func (r *PodGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	case schedv1alpha1.PodGroupPending:
 		if len(pods) >= int(pg.Spec.MinMember) {
 			pgCopy.Status.Phase = schedv1alpha1.PodGroupScheduling
-			fillOccupiedObj(pgCopy, &pods[0])
+			if len(pods) > 0 {
+				fillOccupiedObj(pgCopy, &pods[0])
+			}
 		}
 	default:
 		pgCopy.Status.Running, pgCopy.Status.Succeeded, pgCopy.Status.Failed = getCurrentPodStats(pods)

